### ISSUE TITLE
[a11y]: [Windows High Contrast Mode]: [ProgressBar] The ProgressBar is not visible in High Contrast Mode

### ISSUE THREAD

**Issue by heloiselui** [carbon-design-system/carbon#19356]
### Package

@carbon/react

### Operating System

Windows

### Package version

v1.82.1

### Automated testing tool and ruleset

Manual Verification

### Description

When using Windows High Contrast Mode, the ProgressBrar component is not visible. This affects accessibility, especially for users relying on forced color schemes.

I tested this using Windows system settings and verified across all available high contrast themes. The issue is visible in the video.

Not working:

https://github.com/user-attachments/assets/544703f4-eb0c-404a-8b27-856d953612a5

Working:

https://github.com/user-attachments/assets/8f6e0f47-a644-426a-a170-21f3e3853538

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-progressbar--default

### Steps to reproduce

On Windows:

1. Go to `Settings` > `Accessibility` > `Contrast themes`.
2. Apply any of the high contrast options (e.g., High Contrast `Aquatic`).
3. Open the `React Storybook`.
4. Navigate to the component `ProgressBar` component (e.g., `Default`).
5. Notice that ProgressBar component ir not visible.

Without Windows (using DevTools):

1. Open the `React Storybook`.
2. Open the browser `DevTools` (e.g., Chrome, Edge...).
3. Press` Cmd + Shift + P` to open the command menu.
4. Type “forced”, then select "Emulate CSS media feature forced-colors: active"
5. With this emulation enabled, inspect the `ProgressBar` components again.
6. You should see the same visibility issue as in Windows High Contrast Mode.
> [!WARNING] 
> _Don’t close the DevTools while testing, the emulation only works while DevTools is open._

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
